diffrax/_integrate.py

```diff
@@ -1235,9 +1246,9 @@ def _allocate_output(subsaveat: SubSaveAt) -> SaveState:
             # maximum amount of steps we can possibly take.
             if max_steps is None:
                 raise ValueError(
-                    "`max_steps=None` is incompatible with saving at `steps=True`"
+                    "`max_steps=None` is incompatible with saving at `steps=n`"
                 )
-            out_size += max_steps
+            out_size += max_steps // subsaveat.steps
         if subsaveat.t1 and not subsaveat.steps:
```

Comment:
Note that we have quite a few usecases of `if subsaveat.steps` or `not subsaveat.steps` that should become `if subsaveat.steps != 0` and `if subsaveat.steps == 0`.

Reply:
Although in this particular line I think the logic needs to be something like `if subsaveat.t1 and (max_steps % subsaveat.steps == 0)`, since it may be the case that we need to store an extra `t1` that does not occur in one of our saved steps.